search/search.go

```diff
@@ -25,3 +31,408 @@ type Constraint interface {
 	// path is the path for the column that is constrained
 	path() string
 }
+
+type RowReaderCloser interface {
+	parquet.RowReader
+	io.Closer
+}
+
+func Match(
+	c Constraint,
+	labelFile *parquet.File,
+	labelSchema *parquet.Schema,
+	chunkFile *parquet.File,
+	chunkSchema *parquet.Schema,
+) (RowReaderCloser, error) {
+	labelRowGroups := labelFile.RowGroups()
+	chunkRowGroups := chunkFile.RowGroups()
+
+	joinedSchema := schema.Joined(labelSchema, chunkSchema)
+	if err := c.init(joinedSchema); err != nil {
+		return nil, fmt.Errorf("unable to initialize constraints: %w", err)
+	}
+
+	numRowGroups := len(labelRowGroups)
+
+	rrs := make([]RowReaderCloser, 0, numRowGroups)
+	for i := 0; i != numRowGroups; i++ {
+		ranges, err := c.filter(labelRowGroups[i], false, []rowRange{{from: 0, count: labelRowGroups[i].NumRows() - 1}})
```

Comment:
@MichaHoffmann I'm not sure if I'm getting the first filter call with the entire rowrange right